packages/react/src/use-chat.ui.test.tsx

```diff
@@ -1888,3 +1888,83 @@ describe('test sending additional fields during message submission', () => {
     ),
   );
 });
+
+describe('initialMessages stability', () => {
+  let renderCount = 0;
+
+  const TestComponent = () => {
+    renderCount++;
+    const [derivedState, setDerivedState] = useState<string[]>([]);
+
+    const { messages } = useChat({
+      api: '/api/chat',
+      id: 'test-stability',
+      initialMessages: [
+        {
+          id: 'test-msg-1',
+          content: 'Test message',
+          role: 'user',
+        },
+        {
+          id: 'test-msg-2',
+          content: 'Test response',
+          role: 'assistant',
+        },
+      ],
+    });
+
+    useEffect(() => {
+      setDerivedState(messages.map(m => m.content));
+    }, [messages]);
+
+    if (renderCount > 10) {
+      throw new Error('Excessive renders detected; likely an infinite loop!');
+    }
+
+    return (
+      <div>
+        <div data-testid="render-count">{renderCount}</div>
+        <div data-testid="derived-state">{derivedState.join(', ')}</div>
+        {messages.map(m => (
+          <div key={m.id} data-testid={`message-${m.role}`}>
+            {m.content}
+          </div>
+        ))}
+      </div>
+    );
+  };
+
+  beforeEach(() => {
+    renderCount = 0;
+    render(<TestComponent />);
+  });
+
+  afterEach(() => {
+    cleanup();
+  });
+
+  it('should not cause infinite rerenders when initialMessages is defined and messages is a dependency of useEffect', async () => {
+    await waitFor(() => {
+      expect(screen.getByTestId('message-user')).toHaveTextContent(
+        'Test message',
+      );
+    });
+
+    const initialRenderCount = parseInt(
+      screen.getByTestId('render-count').textContent || '0',
```

Comment:
why `|| '0'`? if this should not happen, can you use `!` instead to fix type error (also below)?